lib/evse_security/crypto/openssl/openssl_provider.cpp

```diff
@@ -23,53 +23,72 @@ typedef struct ossl_lib_ctx_st OSSL_LIB_CTX;
 
 namespace evse_security {
 
-bool is_tpm_key_string(const std::string& private_key_pem) {
-    return private_key_pem.find("-----BEGIN TSS2 PRIVATE KEY-----") != std::string::npos;
+static auto KEY_HEADER_DEFAULT = "-----BEGIN PRIVATE KEY-----";
+static auto KEY_HEADER_TPM2 = "-----BEGIN TSS2 PRIVATE KEY-----";
+
+bool is_custom_key_string(const std::string& private_key_pem) {
+    // If we can't find the standard header it means it's a custom key
+    return private_key_pem.find(KEY_HEADER_DEFAULT) == std::string::npos;
 }
 
-bool is_tpm_key_file(const fs::path& private_key_file_pem) {
+bool is_custom_key_file(const fs::path& private_key_file_pem) {
     if (fs::is_regular_file(private_key_file_pem)) {
         std::ifstream key_file(private_key_file_pem);
         std::string line;
         std::getline(key_file, line);
         key_file.close();
 
-        return line.find("-----BEGIN TSS2 PRIVATE KEY-----") != std::string::npos;
+        // Search for the standard header
+        return line.find(KEY_HEADER_DEFAULT) == std::string::npos;
     }
 
     return false;
 }
 
-#ifdef USING_OPENSSL_3_TPM
+#ifdef USING_CUSTOM_PROVIDER
+
+constexpr bool is_custom_provider_tpm() {
```

Comment:
an alternative
static constexpr bool is_custom_provider_tpm() {
    constexpr const std::string_view s1(CUSTOM_PROVIDER_NAME);
    return s1 == "tpm2";
}

Reply:
looks like the following works:
static constexpr bool is_custom_provider_tpm() {
    return CUSTOM_PROVIDER_NAME == "tpm2";
}

Reply:
> looks like the following works: static constexpr bool is_custom_provider_tpm() { return CUSTOM_PROVIDER_NAME == "tpm2"; }

This works but what it does is that it compares 2 strings that are (const char *). The problem is that the pointers are compared and on certain systems/compiles that do not have string optimizations enabled the expression ("tpm2" == "tpm2") can be either true (if "tpm2" expresses the same string) or it can be false if the strings express different pointers, which can happen. The string_view comparison which forces the character-by-character comparison is the better way.